c++/src/kj/filesystem-disk-test.c++

```diff
@@ -208,16 +210,19 @@ bool isWine() { return false; }
 #endif
 
 static Own<File> newTempFile() {
-  char filename[] = VAR_TMP "/kj-filesystem-test.XXXXXX";
+  const char* tmpDir = getenv("TEST_TMPDIR");
+  auto filename = str(tmpDir != nullptr ? tmpDir : VAR_TMP, "/kj-filesystem-test.XXXXXX");
   int fd;
-  KJ_SYSCALL(fd = mkstemp(filename));
-  KJ_DEFER(KJ_SYSCALL(unlink(filename)));
+  KJ_SYSCALL(fd = mkstemp(filename.begin()));
```

Comment:
Use `.cStr()` instead of `.begin()`... it does the same thing but is more explicit that you're expecting the result to be usable as a C string.

Reply:
done for the second case. Can't in first since cStr() is const.

Reply:
Oh, right, mkstemp() modifies the buffer. `begin()` is appropriate, then.